modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/storage.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {CONSENT_TYPES, checkConsent} from './consent';
+
+export default class Storage {
+	constructor(storage) {
+		this.storage = storage;
+	}
+
+	clear() {
+		return this.storage.clear();
+	}
+
+	getItem(key, type) {
+		if (!checkConsent(type)) {
+			this.removeItem(key);
```

Comment:
We should replicate the same behavior in cookies, if we don't have consent we just don't return the item but we don't remove it

Reply:
Done! Changed it in https://github.com/liferay-frontend/liferay-portal/pull/2658/commits/7d3ccb254f44845797fa7d197ba75f093c60bb19, also added a check for this in a test.